include/meshio/utilities.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2017, Lakshman Anumolu, Pradeep Garigipati
+ * All rights reserved.
+ *
+ * This file is part of MeshIO whose distribution is governed by
+ * the BSD 2-Clause License contained in the accompanying LICENSE.txt
+ * file.
+ */
+
+#ifndef MESHIO_UTILITIES_HPP_
+#define MESHIO_UTILITIES_HPP_
+
+#include <string>
+#include <cmath>
+
+namespace meshio
+{
+
+/**
+ * Extracts filename from path.
+ */
+static inline std::string getFileBaseName(const std::string& pFileName)
```

Comment:
@9prady9 I need a function that returns base name for file with `file_name.obj`, so that I can use it to write `file_name.mtl`. So, this is aimed for writing obj files and its derivative files. This is why I had it like this, any suggestion in this regard? At the end I do need a function that returns basename and it does expect an extension.

I will mention in the documentation too.